test/components/retrievers/test_filter_retriever.py

```diff
@@ -0,0 +1,136 @@
+from typing import Dict, Any, List
+
+import pytest
+
+from haystack import Pipeline, DeserializationError
+from haystack.testing.factory import document_store_class
+from haystack.components.retrievers.filter_retriever import FilterRetriever
+from haystack.dataclasses import Document
+from haystack.document_stores.in_memory import InMemoryDocumentStore
+
+
+@pytest.fixture()
+def sample_docs():
+    en_docs = [
+        Document(content="Javascript is a popular programming language", meta={"lang": "en"}),
+        Document(content="Python is a popular programming language", meta={"lang": "en"}),
+        Document(content="A chromosome is a package of DNA ", meta={"lang": "en"}),
+    ]
+    de_docs = [
+        Document(content="python ist eine beliebte Programmiersprache", meta={"lang": "de"}),
+        Document(content="javascript ist eine beliebte Programmiersprache", meta={"lang": "de"}),
+    ]
+    all_docs = en_docs + de_docs
+    return {"en_docs": en_docs, "de_docs": de_docs, "all_docs": all_docs}
+
+
+@pytest.fixture()
+def sample_document_store(sample_docs):
+    doc_store = InMemoryDocumentStore()
+    doc_store.write_documents(sample_docs["all_docs"])
+    return doc_store
+
+
+class TestFilterRetriever:
+    @classmethod
+    def _documents_equal(cls, docs1: List[Document], docs2: List[Document]) -> bool:
+        return set(d.content for d in docs1) == set(d.content for d in docs2)  # noqa: C401
```

Comment:
```suggestion
        return set(docs1) == set(docs2)
```
I would prefer to compare the entire Documents.
Am I missing something?

Reply:
Hey, Yes, makes sense! 

Switched to comparing docs. Just sorted with content to align docs. (can't directly create set has Documents are not hashable)